src/domain/mongoose_lazy_routing.erl

```diff
@@ -0,0 +1,397 @@
+%% Generally, you should not call anything from this module directly.
```

Comment:
I'd rather see a comment documenting the purpose of this module at the top. Could you add two sentences describing why the laziness is needed and what it means?